ludwig/models/trainer.py

```diff
@@ -601,6 +601,8 @@ def run_evaluation(
         Returns whether the trainer should early stop, based on validation metrics history.
         """
         start_time = time.time()
+        self.callback(lambda c: c.on_eval_start(self, progress_tracker, save_path))
```

Comment:
Just curious: the docstring for `on_eval_start` says it's called from the coordinator, but this invocation doesn't check `is_coordinator()`. Is that intentional?

Reply:
Check the implementation of `self.callback` ;)

Reply:
I missed this detail initially as well. Keeping the `self.is_coordinator` centralized makes sense to avoid code duplication.